sessionctx/variable/session.go

```diff
@@ -1041,6 +1045,25 @@ func (s *SessionVars) setTxnMode(val string) error {
 	return nil
 }
 
+// SetPrevStmtDigest sets the digest of the previous statement.
+func (s *SessionVars) SetPrevStmtDigest(prevStmtDigest string) {
+	s.prevStmtDigest = prevStmtDigest
+}
+
+// GetPrevStmtDigest returns the digest of the previous statement.
+func (s *SessionVars) GetPrevStmtDigest() string {
+	if len(s.prevStmtDigest) > 0 {
+		return s.prevStmtDigest
+	}
+	prevStmt := s.PrevStmt.String()
+	if len(prevStmt) == 0 {
+		return ""
+	}
+	_, digest := parser.NormalizeDigest(s.PrevStmt.String())
```

Comment:
```suggestion
	_, digest := parser.NormalizeDigest(prevStmt)
```
The cost of `s.PrevStmt.String()` is innegligible and, it's result may change:

https://github.com/pingcap/tidb/blob/2cb5eb6060f1ec6ac0282a661907373fae3db87d/executor/adapter.go#L747

Reply:
It seems senseless to normalize a truncated SQL, since even similar statements may result in different digests. How about return `prevStmtDigest` directly?
Only when `commit` comes in once the stmt summary is enabled, it will result in an empty `prevStmtDigest`.

Reply:
If the current SQL is `commit` and `prevStmtDigest` is empty, I can just ignore this `commit`.

Reply:
Sounds reasonable